.ci-builds/.azure-pipelines.yml

```diff
@@ -37,6 +37,9 @@ jobs:
       env:
         CC: clang
         CXX: clang++
+    - script: |
+        cd icu4c/source/test/cintltst && LANG=C LD_LIBRARY_PATH=../../lib:../../tools/ctestfw ./cintltst /tsutil/cloctst/TestCDefaultLocale
```

Comment:
I think it might be good to test for both cases -- so let's add another script entry for LANG=C.UTF-8 and also change the displayName too.